### DISCUSSION TITLE
Questions on stability of the current App integration

### DISCUSSION THREAD

**Discussion by herobaby71** [saleor/apps#328] [General]
Hi Saleor team,

I saw that the saleor-app-sdk is in heavy development at the moment so I just want to get a better understanding of how stable is the app template. Here are my questions:

1/ Can I ask will the saleor-app-template undergo significant changes in the upcoming months?

2/ If so, can we get an estimated timeline on when it would be stable so we can schedule the development of an app.

3/ If not, can I ask what are the expected maintenance costs of a Saleor app in the next eight to twelve months?

Cheers,

**Comment by lkostrowski** [comment]
Hi @herobaby71 , thanks for reaching out. I have moved this discussion to the saleor/apps repo, where we centralize apps-related issues

>saleor-app-sdk is in heavy development

That is true, but I must admin that we don't introduce a lot of breaking changes. What is going to change for sure, is that v1.0.0 will
1. Be moved to the monorepo
2. Will be split into framework-specific packages (eg. app-sdk-nextjs, app-sdk-core, app-sdk-remix etc)

All the code used in app-sdk is open source and it only abstracts logic between client and Saleor. For example, you don't have to manually implement security checks (signature verifications etc) - SDK does it for you and provides you a wrappers around endpoints, so you can focus on your business logic.

The majority of use cases are already done, we still add more of such helpers, but its unlikely to introduce breaking changes here

app-sdk is used in saleor apps and its actively maintained and developed, it is and will be foundation of apps development for Saleor (for JS stack)

Nevertheless I can't promise that sometimes things may break, especially when there is a mismatch between Dashboard and Apps (SDK). SDK is the glue between these services so it should be up to date.


>1/ Can I ask will the saleor-app-template undergo significant changes in the upcoming months?

app-template is only a boilerplate. When you create a new app, you can use it, but you don't have to. There is no logic in app-template that you need. Its rather a boilerplate solution you can kickoff your app. You can check out [examples](https://github.com/saleor/app-examples/tree/main/examples) for more specific cases or [real life production apps](https://github.com/saleor/apps/tree/main/apps) and base on them

app-template philosophy is staying lean, so there are no big changes planned here. We try to keep app-sdk up to date and update it to follow best practices, but we don't bloat it with all features Saleor or SDK offers.

>2/ If so, can we get an estimated timeline on when it would be stable so we can schedule the development of an app.

Unfortunately, we don't have specific deadline on that. Currently we focus on delivering a significant number of apps, that Saleor users can install out of the box. Tooling development is happening as a side effect of that.

However, I don't think you should wait with writing your apps. We have plenty of them already there. Saleor itself is very stable and doesn't introduce breaking changes. If something breaks, we fix it quickly and we can use your feedback. Also everything app-sdk can, your app codebase can do as well, so there is no "ejecting" issues

>3/ If not, can I ask what are the expected maintenance costs of a Saleor app in the next eight to twelve months?

1. Saleor 3x will stay with us for a while. It doesn't introduce breaking changes, so you will not need any updates on the line of installation -> listening to webhooks -> using graphql
2. When you update a minor version of Saleor (3.10->3.11 etc), Dashboard will likely have updated version of app-sdk. We recommend bumping the library also on your side, to ensure they are compatible. We almost don't introduce breaking changes (api-wise), sometimes we dropped some experiments from the past (VercelAPL)
3. We have yet to get support for updating apps - it has to be reinstalled now. We try to do it slowly, but we are still waiting for a deadline. Keep that in mind.

Other things are mainly adding new features, like support for app logo, required Saleor version etc. There are additions to Saleor Core - SDK is just making it easier for developers

Let me know you need any more support or questions
5.

**Reply by herobaby71** [reply]
Thank you
